crawl-ref/source/player.cc

```diff
@@ -3025,6 +3027,22 @@ void level_change(bool skip_attribute_increase)
         learned_something_new(HINT_NEW_LEVEL);
     }
 
+    if (you.char_class == JOB_ARCHAEOLOGIST && you.experience_level >= 3)
+    {
+        int tome_index = -1;
+        for (int i = 0; i < ENDOFPACK; i++)
+            if (you.inv[i].defined() && you.inv[i].is_type(OBJ_MISCELLANY, MISC_DUSTY_TOME))
+                tome_index = i;
+
+        if (tome_index != -1) 
+            archaeologist_read_tome(you.inv[tome_index]);
+        else if(!you.props.exists(ARCHAEOLOGIST_TRIGGER_TOME_ON_PICKUP))
+            mprf("You suddenly remember the dusty tome you brought into the dungeon! You feel able"
+                 " to decipher it now. If only you could remember where you put it..."); 
+
```

Comment:
More lines too long.... the if and the mprf(), and the mprf() should be an mpr().